### ISSUE TITLE
Add tagged releases to match godot releases

### ISSUE THREAD

**Issue by sambler** [godotengine/godot-demo-projects#16]
When the demos were included with the main godot repo we had a set of demos available that matched the features for each godot version released.

It would be clearer if the demos had tagged releases to match each godot release rather than just looking for the godot release date and picking a commit made before the release or a matching branch head. These releases would also match the versioned release of demos available to download at godotengine.org

**Comment by akien-mga** [comment]
That's up for discussion, but I chose not to do so as I believe the demos are not strongly linked to the Godot version (especially not to patch releases like 2.0.3 or 2.1.1).

So instead I chose to create a branch for the current stable version: https://github.com/godotengine/godot-demo-projects/tree/2.1 (currently the same as master)
New demos that would be 2.2-specific (e.g. showcasing VisualScript or the advanced networking API) would go in the master branch only, while bug fixes on existing demos or new improved demos would go in both branches.

The versioned demos available from godotengine.org are therefore not tag-bound, but a snapshot of the 2.1 branch at a given date, see https://downloads.tuxfamily.org/godotengine/demos/2.1/

My intent is that people who downloaded the demos for 2.1 won't have to redownload them for 2.1.1 or 2.1.2, but that people downloading demos 2 months from now would get the latest version from the stable branch.

It's slightly less convenient for Linux packagers, but a snapshot of the 2.1 branch can easily be retrieved with https://github.com/godotengine/godot-demo-projects/archive/2.1/godot-demo-projects-2.1.tar.gz (which is what I use in my Mageia package: http://svnweb.mageia.org/packages/cauldron/godot/current/SPECS/godot.spec?view=markup)

As I mentioned, I'm open for discussion, so if there are strong arguments in favour of tagging releases instead of using this stable branch approach I went for, I might reconsider :)
